shared_model/builders/protobuf/builder_templates/transaction_template.hpp

```diff
@@ -105,19 +107,35 @@ namespace shared_model {
       auto creatorAccountId(
           const interface::types::AccountIdType &account_id) const {
         return transform<CreatorAccountId>([&](auto &tx) {
-          tx.mutable_payload()->set_creator_account_id(account_id);
+          tx.mutable_payload()
+              ->mutable_reduced_payload()
+              ->set_creator_account_id(account_id);
+        });
+      }
+      auto batchMeta(interface::types::BatchType type,
+                     std::vector<interface::types::HashType> hashes) const {
+        return transform<0>([&](auto &tx) {
+          tx.mutable_payload()->mutable_batch()->set_type(
+              static_cast<
+                  iroha::protocol::Transaction::Payload::BatchMeta::BatchType>(
+                  type));
+          for (auto hash : hashes) {
+            tx.mutable_payload()->mutable_batch()->add_tx_hashes(
+                hash.toString());
```

Comment:
`toString` should be used only for debugging purpose, please use `crypto::toBinaryString(hash))` instead